projects/com.oracle.truffle.llvm.runtime/src/com/oracle/truffle/llvm/runtime/types/OpaqueType.java

```diff
@@ -75,4 +75,29 @@ public String toString() {
             return name;
         }
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((name == null) ? 0 : name.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
```

Comment:
well, that happens when someone uses the eclipse codegenerator, without further checking :)